core/chains/evm/client/chain_client.go

```diff
@@ -73,7 +73,10 @@ func NewChainClient(
 		chainID,
 		chainType,
 		"EVM",
-		ClassifySendOnlyError,
+		func(tx *types.Transaction, err error) commonclient.SendTxReturnCode {
+			return ClassifySendError(err, logger.Sugared(logger.Nop()), tx, common.Address{}, chainType.IsL2())
```

Comment:
Why not pass the existing logger param instead of Nop?

Reply:
Passing proper logger would result in partial logs that duplicate proper ones. 
`ClassifySendError` is called inside the `SendTransaction`, that is called by `SendTransactionReturnCode`, which perform logging on its own.
`ClassifySendError` called from the `SendTransaction` does not have access to the address of the source account.